features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/service/DefaultGraph.java

```diff
@@ -163,4 +164,22 @@ private void updateLayout(Collection<Vertex> displayVertices, Collection<Edge> d
         }
         LOG.debug("Created a graph with {} vertices and {} edges", m_displayVertices.size(), m_displayEdges.size());
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        DefaultGraph that = (DefaultGraph) o;
+        return Objects.equals(m_displayVertices, that.m_displayVertices) &&
+                Objects.equals(m_displayEdges, that.m_displayEdges) &&
+                Objects.equals(m_layout, that.m_layout) &&
+                Objects.equals(edgeStatus, that.edgeStatus) &&
+                Objects.equals(vertexStatus, that.vertexStatus);
+    }
+
+    @Override
+    public int hashCode() {
+
+        return Objects.hash(m_displayVertices, m_displayEdges, m_layout, edgeStatus, vertexStatus);
```

Comment:
I am not sure if this will work as expected.

In my opinion we should not include `edgeStatus` and `vertexStatus` for `hashCode` or `equals`.
The layout should be "recalculated" if the displaying vertices or edges have changed.
At the moment if the vertices and edges have not changed, but a status (e.g. one vertex is now red instead of green), the layout would be recalculated, which is not ideal, as the positions of the vertices/edges can basically stay the same, only the UI needs to know about the new state.

Reply:
ok, I changed the hashing: https://github.com/OpenNMS/opennms/blob/173a37e706df019d70e68640217f81820efa8044/features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/VEProviderGraphContainer.java#L255

Reply:
Okay, that is much better. Can you please add a comment, why the hash is build this way and please remove the `equals()` and `hashCode()` methods from the `DefaultGraph` if they are not needed?